### ISSUE TITLE
Report usage for usage-based billing?

### ISSUE THREAD

**Issue by phortx** [PJUllrich/lemon_ex#12]
Is there currently a way to report the usage for usage-based billing?

from https://docs.lemonsqueezy.com/guides/developer-guide/usage-based-billing

```http
POST https://api.lemonsqueezy.com/v1/usage-records

{
  "data": {
    "type": "usage-records",
    "attributes": {
      "quantity": 1,
      "action": "increment"
    },
    "relationships": {
      "subscription-item": {
        "data": {
          "type": "subscription-items",
          "id": "1"
        }
      }
    }
  }
}
```

**Comment by phortx** [comment]
Or at least the possibility to make custom requests?

**Comment by PJUllrich** [comment]
Hey! No, I don't support the usage-based billing yet, but you can always make custom requests with e.g. `LemonEx.Request.get("/usage-records")` which should return what you expect.

**Comment by PJUllrich** [comment]
@phortx just added `UsageRecords` and released a new version `0.2.3` 👍 Hope it helps!

**Comment by phortx** [comment]
What the fork!? Dude, how fast was that 😄 🚀 

Amazing! I will give it a try this week, thank you very much! So impressive, much fast, very wow 🐕  😄

**Comment by PJUllrich** [comment]
Haha, my pleasure!On 25. Jun 2024, at 18:57, Ben Klein ***@***.***> wrote:﻿
What the fork!? Dude, how fast was that 😄 🚀
Amazing! I will give it a try this week, thank you very much! So impressive, much fast, very wow 🐕  😄

**Comment by phortx** [comment]
Hi,

I think there is still a bug:

https://github.com/PJUllrich/lemon_ex/blob/c2249c66ea5d007324e930e2e8b0745dfa3186dc/lib/lemon_ex/usage_records/usage_records.ex#L11

This line should be

```elixir
"subscription-item": %{
```

instead of

```elixir
subscription_item: %{
```

According to https://docs.lemonsqueezy.com/api/usage-records#create-a-usage-record

**Comment by phortx** [comment]
I created a PR to fix that :) With that fix, everything works well. Thank you!

**Comment by PJUllrich** [comment]
merged! Thank you! ❤️

**Comment by PJUllrich** [comment]
Published as `0.2.4`

**Comment by phortx** [comment]
Perfect, thank you :)
